js/ui/grid_core/ui.grid_core.keyboard_navigation.js

```diff
@@ -94,59 +94,55 @@ function isCellInHeaderRow($cell) {
 const KeyboardNavigationController = core.ViewController.inherit({
     // #region Initialization
     init: function() {
-        const that = this;
-
-        if(that.isKeyboardEnabled()) {
+        this._dataController = this.getController('data');
+        this._selectionController = this.getController('selection');
+        this._editingController = this.getController('editing');
+        this._headerPanel = this.getView('headerPanel');
+        this._columnsController = this.getController('columns');
+        this._editorFactory = this.getController('editorFactory');
+
+        if(this.isKeyboardEnabled()) {
             accessibility.subscribeVisibilityChange();
-            that._dataController = that.getController('data');
-            that._selectionController = that.getController('selection');
-            that._editingController = that.getController('editing');
-            that._headerPanel = that.getView('headerPanel');
-            that._columnsController = that.getController('columns');
-
-            that._fastEditingStarted = false;
-            that._focusedCellPosition = {};
-            that._canceledCellPosition = null;
-
-            that.getController('editorFactory').focused.add(function($element) {
-                that.setupFocusedView();
-
-                if(that._isNeedScroll) {
-                    if($element.is(':visible') && that._focusedView && that._focusedView.getScrollable) {
-                        that._focusedView._scrollToElement($element);
-                        that._isNeedScroll = false;
+
+            this._updateFocusTimeout = null;
+            this._fastEditingStarted = false;
+            this._focusedCellPosition = {};
+            this._canceledCellPosition = null;
+
+            const elementFocused = ($element) => {
```

Comment:
Is this changes in init necessary?

Reply:
The **focus** event handler was declared as a separate function. I don't see issues with this solution.